### ISSUE TITLE
condition of main dex exceeds 65k

### ISSUE THREAD

**Issue by bgeradz** [KeepSafe/dexcount-gradle-plugin#243]
If a project already exceeds 65k methods in it's main dex it won't assemble and the plugin won't help to find the problem.

I have managed to write some gradle that creates dex files for classes that would go to main dex.

It would be nice if the plugin could do it's analysis on a given arbitrary set of dex files, not just those created by successful assembly and only after successful build.

**Comment by benjamin-bader** [comment]
I'm not sure that I understand what you mean.  Let me rephrase, and tell me if I got it right.  You have a Gradle project that compiles Java sources into .dex files, and you would like this plugin to count the method references inside of those dex files?

You could write this without too much effort - `DexFile.extractDexData(File)` will work with .dex files as well as regular APKs.  You'd need to set up the Gradle task yourself; look at the Plugin class to see how we currently do it.

As a side note, you opened the issue by stating that a project over the limit won't assemble.  This isn't quite correct - you can always enable multidex, and your project will build just fine.  The plugin will count the resulting APK without issue.

**Comment by bgeradz** [comment]
Thank you very much for response!
I will investigate further.

As for the issue, there is no resulting apk because 65k limit is exceeded,
that's the point and the problem

**Comment by benjamin-bader** [comment]
Of course there's no APK if the 65k limit is exceeded - you have to enable multidex in your build for it to work: https://developer.android.com/studio/build/multidex

Is there a reason this wouldn't be applicable to you?

**Comment by benjamin-bader** [comment]
Hi there, did you have any luck getting multidex working?

**Comment by bgeradz** [comment]
Hi
Nope, I've gone the other way, not using the plugin.
